src/models.rs

```diff
@@ -433,6 +434,14 @@ impl<'a> Uses<'a> {
             Uses::Repository(repo) => repo.git_ref.is_none(),
         }
     }
+
+    pub(crate) fn unpinned_sha(&self) -> bool {
+        let regex = Regex::new("^[a-fA-F0-9]{40}$").unwrap();
+        match self {
+            Uses::Docker(docker) => ! docker.hash.is_none(),
+            Uses::Repository(repo) => repo.git_ref.is_none() || ! regex.is_match(repo.git_ref.unwrap()),
```

Comment:
If I'm following the logic here correctly, I think this can reuse the existing `ref_is_commit()` helper:

```suggestion
            Uses::Repository(repo) => !repo.ref_is_commit(),
```

(and then you can remove the `Regex` object :slightly_smiling_face:)